src/Common/Extensions/UrlExtensions.cs

```diff
@@ -0,0 +1,65 @@
+namespace Passwordless.Common.Extensions;
+
+public static class UrlExtensions
+{
+    public static bool IsLocalUrl(this string? url)
+    {
+        if (string.IsNullOrEmpty(url))
+        {
+            return false;
+        }
```

Comment:
Extension methods typically shouldn't be callable on `null` instances (similar to member methods) to avoid confusion